action.yml

```diff
@@ -40,16 +52,13 @@ runs:
           curl --silent --show-error --fail --location --output $TARGET "$LOCATION"
           tar -xf $TARGET
         fi
-      shell: bash
     - name: Lint files
+      shell: bash
       run: |
         set -u
-        set +e
         if [ -z ${{ inputs.config }} ]; then
           CONFIG=""
         else
           CONFIG="--config ${{ inputs.config }}"
         fi
-        ./kube-linter $CONFIG lint ${{ inputs.directory }} 2>&1 | tee -a ${{ inputs.output-file }}
-        exit ${PIPESTATUS[0]}
-      shell: bash
+        ./kube-linter $CONFIG lint ${{ inputs.directory }} --format ${{ inputs.format }} | tee ${{ inputs.output-file }}
```

Comment:
Why did you remove the `-a`?

Reply:
How come the `set +e` with `exit ${PIPESTATUS[0]}` was necessary before but isn't now? Or was it just never necessary before?

Reply:
> Why did you remove the `-a`?

Appending log file only makes sense when the output format is `plain`. However, users can run the action multiple times with different formats but using the default output file name. In this case `plain` output would get appended to `json` or `sarif` messing up the output file.

> How come the `set +e` with `exit ${PIPESTATUS[0]}` was necessary before but isn't now?

I think the idea was to return exit code from `kube-linter` only and ignore exit code from `tee`.
GitHub starts bash commands with `set -eo pipefail` already by default so the script previously had `set +e` to allow capturing `kube-linter`-s exit code. Then it would return only this exit code by using `${PIPESTATUS[0]}`.
First, I moved `set +e` closer to where `kube-linter` was started. Then I played with self-test and saw that a case where `output-file` is in not-existing directory does not fail. See https://github.com/stackrox/kube-linter-action/runs/3743081013#step:3:45
Therefore, the right thing is to let any command of the `kube-linter | tee` pipeline fail the action. This is simply achieved by relying on GitHub's defaults.